main/exercise/multipleAnswerTrueFalseDegreeCertainty.php

```diff
@@ -0,0 +1,1220 @@
+         * @param FormValidator $form
+         * @param Exercise      $exercise
+         */
+        public function processAnswersCreation($form, $exercise)
+        {
+            $questionWeighting = $nbrGoodAnswers = 0;
+            $objAnswer = new Answer($this->id);
+            $nbAnswers = $form->getSubmitValue('nb_answers');
+
+            $courseId = api_get_course_int_id();
+
+            $correct = [];
+            $options = Question::readQuestionOption($this->id, $courseId);
+
+            if (!empty($options)) {
+                foreach ($options as $optionData) {
+                    $id = $optionData['id'];
+                    unset($optionData['id']);
+                    Question::updateQuestionOption($id, $optionData, $courseId);
+                }
+            } else {
+                for ($i = 1; $i <= 8; $i++) {
+                    $lastId = Question::saveQuestionOption($this->id, $this->options[$i], $courseId, $i);
+                    $correct[$i] = $lastId;
+                }
+            }
+
+            /* Getting quiz_question_options (true, false, doubt) because
+              it's possible that there are more options in the future */
+
+            $newOptions = Question::readQuestionOption($this->id, $courseId);
+
+            $sortedByPosition = [];
+            foreach ($newOptions as $item) {
+                $sortedByPosition[$item['position']] = $item;
+            }
+
+            /* Saving quiz_question.extra values that has the correct scores of
+              the true, false, doubt options registered in this format
+              XX:YY:ZZZ where XX is a float score value. */
+            $extraValues = [];
+
+            for ($i = 1; $i <= 3; $i++) {
+                $score = trim($form->getSubmitValue('option['.$i.']'));
+                $extraValues[] = $score;
+            }
+            $this->setExtra(implode(':', $extraValues));
+
+            for ($i = 1; $i <= $nbAnswers; $i++) {
+                $answer = trim($form->getSubmitValue('answer['.$i.']'));
+                $comment = trim($form->getSubmitValue('comment['.$i.']'));
+                $goodAnswer = trim($form->getSubmitValue('correct['.$i.']'));
+                if (empty($options)) {
+                    //If this is the first time that the question is created when change the default values from the form 1 and 2 by the correct "option id" registered
+                    $goodAnswer = $sortedByPosition[$goodAnswer]['id'];
+                }
+                $questionWeighting += $extraValues[0]; //By default 0 has the correct answers
+                $objAnswer->createAnswer($answer, $goodAnswer, $comment, '', $i);
+            }
+
+            // saves the answers into the data base
+            $objAnswer->save();
+
+            // sets the total weighting of the question
+            $this->updateWeighting($questionWeighting);
+            $this->save($exercise);
+        }
+
+        /**
+         * @param int   $feedbackType

```

Comment:
Une petite description serait bienvenue: "Returns header of the results table for certainty level questions" ?